platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -418,7 +418,12 @@ OtaPalStatus_t otaPal_CreateFileForRx( OtaFileContext_t * const C )
 
     if( C != NULL )
     {
-        if( C->pFilePath != NULL )
+        if( ( C->pFilePath != NULL ) && ( strlen( ( const char * ) C->pFilePath ) + 1U > OTA_FILE_PATH_LENGTH_MAX ) )
```

Comment:
hmm the +1 to account for null terminator may not be needed - at least in my mind OTA_FILE_PATH_LENGTH_MAX should be able to store a path that is that long (not including the null terminator), but if this macro defines a specific buffer size and that buffer is expected to be null-terminated this is fine.